src/Umbraco.Web.UI.Client/src/common/services/varianthelper.service.js

```diff
@@ -0,0 +1,61 @@
+﻿/**
+* @ngdoc service
+* @name umbraco.services.variantHelper
+* @description A helper service for dealing with variants
+**/
+function variantHelper() {
+    /**
+     * Returns the id for this variant
+     * @param {any} variant
+     */
+    function getId(variant) {
```

Comment:
Ideally some unit tests would be made for these methods.Angular services are the easiest to test so should be relatively easy to setup.